### ISSUE TITLE
Add support for common aggregate functions

### ISSUE THREAD

**Issue by zlwaterfield** [supabase/postgrest-js#206]
# Feature request

There are a bunch of commonly used Postgres methods that should be added:
 - `COUNT(*)`
 - `AVG(number)`
 - `MIN(number)`
 - `MAX(number)`

**Comment by PH4NTOMiki** [comment]
@kiwicopple what's up with this, is it planned soon?
Thanks

**Comment by steve-chavez** [comment]
@PH4NTOMiki Progress for this is being tracked at https://github.com/PostgREST/postgrest/issues/915#issuecomment-1002232582

**Comment by steve-chavez** [comment]
An example of how this can be done now:

```sql
create or replace function max_value() returns int as $$
  select max(my_column) from my_table;
$$ language sql;
```

```js
const { data, error } = await supabase
  .rpc('max_value')
```

References:

- https://supabase.com/docs/reference/javascript/rpc
- https://stackoverflow.com/questions/72814312/max-aggregate-function-syntax-to-be-called-by-the-supabase-client/72823179#72823179

**Comment by jdgamble555** [comment]
'Group By' is provided by postgres. Could we add this feature?

 It should be relatively simple.

This is also very important because you can't subscribe to Views or Procedures. There is no work around for that.  

J

**Comment by felixcheruiyot** [comment]
Looking for this one too.

**Comment by JRB-y** [comment]
Is there a plan for the 'distinct' function ?

**Comment by rawestmoreland** [comment]
+1 for a distinct function. The supabase AI references it when searching docs, but the method doesn't exist.

**Comment by JRB-y** [comment]
> +1 for a distinct function. The supabase AI references it when searching docs, but the method doesn't exist.

Yea the AI confused me too.

**Comment by bizrockman** [comment]
+1 for common postgres functions

**Comment by laubonghaudoi** [comment]
Any update on this?

**Comment by aferrugento** [comment]
+1 this would be really useful

**Comment by Shooteger** [comment]
Hey, I need max() or min() would appreciate if someone smarter than me integrates it.

**Comment by revskill10** [comment]
Is there any ETA for this "important" feature ?

**Comment by jdgamble555** [comment]
> Is there any ETA for this "important" feature ?

It looks like it is in the latest version:

https://postgrest.org/en/stable/references/api/aggregate_functions.html

J

**Comment by revskill10** [comment]
@jdgamble555 Does supabase supports it in latest js-sdk yet ?

**Comment by jdgamble555** [comment]
Looks like soon, not sure - https://supabase.com/blog/postgrest-12

J

**Comment by ConProgramming** [comment]
Awesome to see this official! https://supabase.com/blog/postgrest-aggregate-functions.

But Typescript support still missing. So for example get the error `ParserError<"Expected identifier at `), label`">[]`.

Would love to see fixed quick like #459 was.

**Comment by steve-chavez** [comment]
Closing since support was already announced https://supabase.com/blog/postgrest-aggregate-functions :tada: 

The typescript support can be tracked on another issue.
